thunder/dynamo/utils.py

```diff
@@ -0,0 +1,310 @@
+from enum import Enum, auto
+import dataclasses
+from typing import List, Dict, Optional, Tuple, Set
+from collections.abc import Callable
+import pprint
+import itertools
+import copy
+import inspect
+
+import torch
+from torch.fx._lazy_graph_module import _LazyGraphModule
+import warnings
+from collections.abc import Mapping
```

Comment:
(personal memo): the difference between `Mapping` and `dict` is explained in https://mypy.readthedocs.io/en/stable/builtin_types.html#generic-types